server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -682,13 +681,6 @@ public void batchDelete(List<Pool> pools) {
         for (Pool pool : pools) {
             currentSession().delete(pool);
         }
-        for (Pool pool : pools) {
```

Comment:
The risk is that if pool.entitlements has been initialized before this PoolCurator.batchDelete is called, we might get ConstraintViolation exceptions on Postgre.